frame/nomination-pools/src/tests.rs

```diff
@@ -3148,6 +3153,10 @@ mod withdraw_unbonded {
 				current_era += 3;
 				CurrentEra::set(current_era);
 
+				// set metadata to check that it's being removed on dissolve
+				assert_ok!(Pools::set_metadata(Origin::signed(900), 1, vec![1, 1]));
```

Comment:
Instead of this, make sure that the default pool created in the `ExtBuilder` has some metadata.

Reply:
And check it in the `basic_setup_works`.